docs/guide/styles.md

```diff
@@ -1,15 +1,320 @@
+
+The same units may also be used to set limits on a dimension. The following styles set minimum and maximum sizes and can accept any of the values used in width and height.
+
+- [min-width](../styles/min_width.md) sets a minimum width.
+- [max-width](../styles/max_width.md) sets a maximum width.
+- [min-height](../styles/min_height.md) sets a minimum height.
+- [max-height](../styles/max_hright.md) sets a maximum height.
+
+### Padding
+
+Padding adds space around your content which can aid readability. Setting [padding](../styles/padding.md) to an integer will add that number additional rows and columns around the content area. The following example sets padding to 2:
+
+```python title="padding01.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/padding01.py"
+```
+
+Notice the additional space around the text:
+
+```{.textual path="docs/examples/guide/styles/padding01.py"}
+```
+
+You can also set padding to a tuple of two integers which will apply padding to the top/bottom and left/right edges. The following example sets padding to `(2, 4)` which adds two rows to the top and bottom of the widget, and 4 columns to the left and right of the widget.
+
+```python title="padding02.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/padding02.py"
+```
+
+Compare the output of this example to the previous example:
+
+```{.textual path="docs/examples/guide/styles/padding02.py"}
+```
+
+You can also set padding to a tuple of four values which applies padding to each edge individually. The first value it the padding for the top of the widget, followed by the right of the widget, then bottom, then left.
+
+### Border 
+
+The [border](../styles/border.md) style draws a border around a widget. To add a border set `syles.border` to a tuple of two values. The first value is the border type, which should be a string. The second value is the border color which will accept any value that works with  [color](../styles/color.md) and [background](../styles/background.md).
+
+The following example adds a border around a widget:
+
+```python title="border01.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/border01.py"
+```
+
+Here is the result:
+
+```{.textual path="docs/examples/guide/styles/border01.py"}
+```
+
+There are many other border types. Run the following from the command prompt to preview them.
+
+
+```bash
+textual borders
+```
+
+### Outline
+
+[Outline](../styles/outline.md) is similar to border and is set in the same way. The difference is that outline will not change the size of the widget, and may overlap the content area. The following examples sets an outline on a widget:
+
+```python title="outline01.py" hl_lines="22"
+--8<-- "docs/examples/guide/styles/outline01.py"
+```
+
+Notice how the outline overlaps the text in the widget.
+
+```{.textual path="docs/examples/guide/styles/outline01.py"}
+```
+
+Outline can be useful to emphasize a widget, but be mindful that it may obscure your content.
+
+### Box sizing
```

Comment:
I remember having difficulty grokking the box-sizing description in the styles reference, but this section is great. Very well explained.